sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/OptimizerLoggingSuite.scala

```diff
@@ -49,12 +49,19 @@ class OptimizerLoggingSuite extends PlanTest {
       case event => Seq(
         "Applying Rule",
         "Result of Batch",
-        "has no effect").exists(event.getRenderedMessage().contains)
+        "has no effect",
+        "Metrics of Executed Rules").exists(event.getRenderedMessage().contains)
```

Comment:
this only exists if `Result of Batch` exists?

Reply:
The metrics will be reported at `RuleExecutor.execute` level. I double checked the code and found I logged it at the wrong place within `batches.foreach` loop.  Fixed it by abc864ef7e41d65932060436331a57db8d77fbe7.